### ISSUE TITLE
CI: yamlRestTest fail for DataStreams

### ISSUE THREAD

**Issue by costin** [elastic/elasticsearch#62613]
The build fails due to warnings appearing in the response due to creation of templates with patterns that clash with other templates. The seed is important - running it without (thus a different seed) made the build pass for me.

**Build scan**:
https://gradle-enterprise.elastic.co/s/lh3h7teyduuh6
https://gradle-enterprise.elastic.co/s/d3rweuw4zwqik

**Repro line**:
``
./gradlew ':x-pack:plugin:integTestRunner' --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=security/authz/50_data_streams/Test that requests not supporting data streams do not include data streams among authorized indices}" \
  -Dtests.seed=D58F071C4CBE0212 \
  -Dtests.security.manager=true \
  -Dtests.locale=tr \
  -Dtests.timezone=America/Bogota \
  -Druntime.java=8 \
  -Dtests.rest.blacklist=getting_started/10_monitor_cluster_health/*
``

**Reproduces locally?**: Yes

**Applicable branches**: master and 7.x, 7.9

**Failure history**:

**Failure excerpt**:

```
java.lang.AssertionError: Failure at [security/authz/50_data_streams:63]: got unexpected warning header [
	299 Elasticsearch-8.0.0-SNAPSHOT-2ec7260e2a2fe89eeae49e8ba5cf329e7798aedd "index template [my-template1] has index patterns [s*, easy-data-stream1, create-doc-data-stream1, write-data-stream1] matching patterns from existing older templates [global] with patterns (global => [*]); this template [my-template1] will take precedence during new index creation"
]
```
```
java.lang.AssertionError: Failure at [security/authz/50_data_streams:63]: got unexpected warning header [
	299 Elasticsearch-7.9.2-SNAPSHOT-14fe9cfbbc052ef813b9415596c88f9799db1600 "index template [my-template1] has index patterns [s*, easy-data-stream1, create-doc-data-stream1, write-data-stream1] matching patterns from existing older templates [global] with patterns (global => [*]); this template [my-template1] will take precedence during new index creation"
]
```

**Comment by costin** [comment]
See also https://github.com/elastic/elasticsearch/issues/62611

**Comment by martijnvg** [comment]
This is fixed via #62610,
I didn't know this issue was opened.
